### PR TITLE
Display only the version banner for browser versions that crash on App render

### LINKED ISSUE #1511: Meta: Improve Cross-Browser Compatibility
This is a meta-issue tracking work happening to improve cross-browser compatibility. Currently, we only support Chrome (v76+?).

- [ ] Edge: I don't believe anyone has tested, but since it uses Blink it might work?
- [ ] Safari, and all browsers on iOS/iPadOS (which are forced to use Webkit) (https://github.com/foxglove/studio/issues/1422)
- [ ] Firefox

### PR DESCRIPTION
**Pull request by esthersweon**
**User-Facing Changes**
User should be able to see partly working app on browsers outside of Chrome (supported) and Safari 14 (not functional). 

Firefox (not supported, but partly working app) vs. Safari 14 (app not functional):
Screen Shot 2021-07-20 at 2 47 57 PM

**Description**
Does a more generic check for any browser that does not recognize `BigInt64Array` / `BigUint64Array` to render just the banner (i.e. don't load the app at all, as whole page including banner will crash).

Resolve #1511.

### CODE REVIEW

**Comment by amacneil** [comment]
Is `BigUint64Array` the only browser-compatibility issue we have at this point?

Maybe I am old school but it feels more sensible to whitelist a set of browsers we have tested with (only Chrome at this stage), given how many different things could potentially cause issues?

cc @jhurliman @defunctzombie @jtbandes @esthersweon

File: web/src/VersionBanner.tsx [hunk 1]
```diff
@@ -51,11 +62,10 @@ const VersionBanner = function ({
         )}
       </div>
 
-      <div style={{ paddingTop: "20px" }}>
+      <div style={{ paddingTop: "10px" }}>
         <a href="https://www.google.com/chrome/" target="_blank" rel="noreferrer">
           <button>{fixText}</button>
```
Comment (jtbandes) [inline web/src/VersionBanner.tsx:67]: can we use one of the FluentUI Button classes? our intent is to remove the global css that makes buttons look like they currently do

File: web/src/VersionBanner.tsx [hunk 2]
```diff
@@ -32,16 +39,20 @@ const VersionBanner = function ({
...
   const fixText = isChrome ? "Update Chrome" : "Download Chrome";
-  const continueText = isChrome
-    ? "Continue with unsupported version"
-    : "Continue with unsupported browser";
 
   return (
     <StyledBanner>
       <div>
+        {isDismissable ? (
+          <StyledIconWrapper onClick={() => setShowBanner(false)}>
```
Comment (jtbandes) [inline web/src/VersionBanner.tsx:50]: I think @2metres recently added a HoverableIconButton that you might want to use for this?

Comment (2metres) [reply web/src/VersionBanner.tsx:56]: Currently wrapped up in my PR #1466 , hopefully merging today.

File: web/src/VersionBanner.tsx (hunk 1 above)
Comment (defunctzombie) [reply web/src/VersionBanner.tsx:73]: 👍 We should not use native elements.

**Comment by 2metres** [comment]
Would the: [MessageBar](https://developer.microsoft.com/en-us/fluentui#/controls/web/messagebar) from fluentui work here?

File: web/src/VersionBanner.tsx (hunk 2 above)
Comment (esthersweon) [reply web/src/VersionBanner.tsx:59]: Discussed; will not need.

**Comment by esthersweon** [comment]
> Is `BigUint64Array` the only browser-compatibility issue we have at this point?
> 
> Maybe I am old school but it feels more sensible to whitelist a set of browsers we have tested with (only Chrome at this stage), given how many different things could potentially cause issues?

@amacneil - There are 3 categories of browsers that we want to have different UIs rendered for: 1) supported browsers (i.e. Chrome v76+), 2) not supported, but somewhat working, browsers (e.g. Chrome below v76, Safari 15, Firefox, etc.), and 3) completely broken browsers (e.g. Safari 14).

Discerning category 1 is easy enough; this check for `BigUint64Array` is an easy way to distinguish between categories 2 and 3 without having to parse the user agent for multiple browsers and to check against multiple corresponding version numbers (e.g. Safari - 15, Firefox - 91, etc.).

**Comment by amacneil** [comment]
@esthersweon thanks for clarification, makes sense. When I wrote that comment earlier today I didn't realize we had two separate behaviors for "completely broken" vs "unsupported" browsers. I think the solution we landed on is good.

**Pull request merged**